broker/api_test.go

```diff
@@ -376,17 +378,137 @@ var _ = Describe("Broker API", func() {
+			fakeComposeClient.CreateDeploymentWhitelistReturnsOnCall(0, &composeapi.Recipe{ID: "id-for-1.1.1.1", Status: "complete"}, []error{})
+			fakeComposeClient.CreateDeploymentWhitelistReturnsOnCall(1, &composeapi.Recipe{ID: "id-for-2.2.2.2", Status: "complete"}, []error{})
+			fakeComposeClient.CreateDeploymentWhitelistReturnsOnCall(2, &composeapi.Recipe{ID: "id-for-3.3.3.3", Status: "complete"}, []error{})
+
+			resp := DoRequest(brokerAPI, NewRequest(
+				"PUT",
+				"/v2/service_instances/"+newInstanceID,
+				strings.NewReader(fmt.Sprintf(`{
+					"service_id": "%s",
+					"plan_id": "%s",
+					"organization_guid": "test-organization-id",
+					"space_guid": "space-id",
+					"parameters": {
+						"restore_from_latest_snapshot_of": "%s"
+					}
+				}`, service.ID, service.Plans[0].ID, oldInstanceID)),
+				cfg.Username,
+				cfg.Password,
+				UriParam{Key: "accepts_incomplete", Value: "true"},
+			))
+			Expect(resp.Code).To(Equal(202))
+			body := ReadResponseBody(resp.Body)
+			Expect(body).To(MatchOperationJSON(`
+			{
+			  "recipe_id":"provision-recipe-id",
+			  "type":"provision",
+			  "whitelist_recipe_ids":["id-for-1.1.1.1","id-for-2.2.2.2","id-for-3.3.3.3"]
+			}
+			`))
+
+			By("getting the existing deployment")
+			Expect(fakeComposeClient.GetDeploymentByNameArgsForCall(0)).To(Equal(oldInstanceName))
+
+			By("listing its backups")
+			Expect(fakeComposeClient.GetBackupsForDeploymentArgsForCall(0)).To(Equal(oldInstanceID))
+
+			By("creating a new deployment by restoring the newest backup")
+			expectedRestoreBackupParams := composeapi.RestoreBackupParams{
+				DeploymentID: oldInstanceID,
+				BackupID:     newestBackupID,
+				Name:         newInstanceName,
 				Datacenter:   broker.ComposeDatacenter,
-				DatabaseType: "fakedb",
-				Units:        1,
 				SSL:          true,
-				WiredTiger:   false,
-				Version:      "",
-				ClusterID:    "",
 			}
-			Expect(fakeComposeClient.CreateDeploymentArgsForCall(0)).To(Equal(expectedDeploymentParams))
+			Expect(fakeComposeClient.RestoreBackupArgsForCall(0)).To(Equal(expectedRestoreBackupParams))
+
+			By("setting CustomerBillingCode on the new deployment")
+			expectedPatchDeploymentParams := composeapi.PatchDeploymentParams{
+				DeploymentID:        "2",
+				CustomerBillingCode: "space-id",
+			}
+			Expect(fakeComposeClient.PatchDeploymentArgsForCall(0)).To(Equal(expectedPatchDeploymentParams))
+		})
+
+		It("prevents provisioning an instance from a backup owned by someone else", func() {
+			oldInstanceID := "123467"
+			oldInstanceName, err := broker.MakeInstanceName(cfg.DBPrefix, oldInstanceID)
+			Expect(err).ToNot(HaveOccurred())
+			newestBackupID := "xyz"
+
+			fakeComposeClient.GetDeploymentByNameReturns(&composeapi.Deployment{ID: oldInstanceID, Name: oldInstanceName, ProvisionRecipeID: "provision-recipe-id", CustomerBillingCode: "not-my-space-id"}, nil)
```

Comment:
Could you break this struct out to a separate line for each field so that it's easier to see that the space ID is different? Also the similar line in the preceding test.

Reply:
fixed